holoviews/plotting/mpl/element.py

```diff
@@ -637,8 +640,13 @@ class OverlayPlot(LegendPlot, GenericOverlayPlot):
     _passed_handles = ['fig', 'axis']
 
     def __init__(self, overlay, ranges=None, **params):
-        if overlay.traverse(lambda x: x, (Element3D,)):
-            params['projection'] = '3d'
+        if 'projection' not in params:
+            projs = self._deep_options(overlay, 'plot', ['projection'],
+                                       [Element])['projection']
+            if len(set(projs)) > 1:
+                raise Exception("A single axis may only be assigned one projection type")
```

Comment:
Is this check happening twice? There is an  ``if len(set(opts['projection'])) > 1:`` check in ``MPLPlot`` as well. I'm trying to think if there is one place that could handle both...